Bug: RabbitMQ Broker's Connection

**Describe the bug**
If you create a broker. Connect and then create publisher, an error occurs when trying to send a message

**How to reproduce**
Include source code:

```python
from faststream.rabbit import RabbitBroker, RabbitQueue

artifact_download = RabbitQueue("artifact_download", True)

async def register_download_artifact(artifact):
    broker = RabbitBroker(url="")
    await broker.connect()
    publisher = broker.publisher(artifact_download)
    await publisher.publish(
        message=artifact,
        persist=True,
    )
```

**Expected behavior**
I expect that the message has been sent

**Observed behavior**
```python
  File "/app/build_downloader/server/api_v1/artifacts/dependencies.py", line 173, in register_download_artifact
    await publisher.publish(
  File "/usr/local/lib/python3.12/site-packages/faststream/rabbit/publisher/usecase.py", line 258, in publish
    assert self._producer, NOT_CONNECTED_YET  # nosec B101
           ^^^^^^^^^^^^^^
AssertionError: Please, `connect()` the broker first.
```
The problem is solved if you connect the broker after creating publisher, and not before creating
```python
broker = RabbitBroker(url="")
publisher = broker.publisher(artifact_download)
await broker.connect()
```
but then the error message is wrong: _Please, connect() the broker **first**._

**Environment**
Running FastStream 0.5.30 with CPython 3.12.3 on Linux

--- a/faststream/__about__.py
+++ b/faststream/__about__.py
@@ -1,5 +1,5 @@
 """Simple and fast framework to create message brokers based microservices."""
 
-__version__ = "0.5.33"
+__version__ = "0.5.34"
 
 SERVICE_NAME = f"faststream-{__version__}"

--- a/faststream/broker/core/usecase.py
+++ b/faststream/broker/core/usecase.py
@@ -285,7 +285,7 @@ class BrokerUsecase(
 
     def publisher(self, *args: Any, **kwargs: Any) -> "PublisherProto[MsgType]":
         pub = super().publisher(*args, **kwargs)
-        if self.running:
+        if self.running or self._connection is not None:
             self.setup_publisher(pub)
         return pub
 

--- a/tests/brokers/base/publish.py
+++ b/tests/brokers/base/publish.py
@@ -579,6 +579,12 @@ class BrokerPublishTestcase(BaseTestcaseConfig):
         assert not mock.called
 
     @pytest.mark.asyncio
+    async def test_publisher_after_connect(self, queue: str):
+        async with self.patch_broker(self.get_broker()) as br:
+            # Should pass without error
+            await br.publisher(queue).publish(None)
+
+    @pytest.mark.asyncio
     async def test_publisher_after_start(
         self,
         queue: str,

